### PR TITLE
[FEAT] Treasure Island Utilities

### PR DESCRIPTION
**Pull request by HBandesh**
# Description

In this PR, I have added treasure island utilities: Tiki Totem and Lunar Calendar. If Tiki Totem is placed, give a bonus +0.1 wood when chopping trees. If Lunar Calendar is placed, give a bonus 10% speed to crops (not fruit)

https://user-images.githubusercontent.com/35486909/215610872-b3b1a89f-2fe5-4db5-90c3-813e5cb4cd81.mov

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/events/landExpansion/plant.test.ts [hunk 1]
```diff
@@ -525,6 +525,47 @@ describe("plant", () => {
     expect(plantedAt).toBe(dateNow - carrotTime * 0.15);
   });
 
+  it("grows faster if Lunar calender is placed", () => {
```
Comment (Ivan-Ricardo-Lopes) [inline src/features/game/events/landExpansion/plant.test.ts:528]:
```suggestion
  it("grows faster if Lunar calendar is placed.", () => {
```

File: src/features/game/events/landExpansion/plant.test.ts [hunk 2]
```diff
@@ -612,6 +653,27 @@ describe("getCropTime", () => {
 
     expect(time).toEqual(60 * 60 * 0.8);
   });
+
+  it("applies a 10% speed boost with Lunar Calenda placed", () => {
```
Comment (Ivan-Ricardo-Lopes) [inline src/features/game/events/landExpansion/plant.test.ts:657]:
```suggestion
  it("applies a 10% speed boost with Lunar Calendar placed.", () => {
```

File: src/features/game/types/treasure.ts [hunk 3]
```diff
@@ -84,3 +84,17 @@ export const BEACH_BOUNTY_TREASURE: Record<BeachBountyTreasure, BeachBounty> = {
...
 };
+
+export type PlaceableTreasures = BoostTreasure;
+
+export const PLACEABLE_TREASURES: Record<
+  PlaceableTreasures,
+  { description: string }
+> = {
+  "Lunar Calendar": {
+    description: "A charm made from crops.",
```
Comment (c-py) [inline src/features/game/types/treasure.ts:95]:
```suggestion
    description: "Crops now follow the lunar cycle! 10% increase to crop growth speed.",
```

File: src/features/game/types/treasure.ts [hunk 4]
```diff
@@ -84,3 +84,17 @@ export const BEACH_BOUNTY_TREASURE: Record<BeachBountyTreasure, BeachBounty> = {
...
+  },
+  "Tiki Totem": {
+    description: "A charm made of wood.",
```
Comment (c-py) [inline src/features/game/types/treasure.ts:98]:
```suggestion
    description: "The Tiki Totem adds 0.1 wood to every tree you chop.",
```

**Review by c-py — CHANGES_REQUESTED** [review]
Looking good mate! Just one feedback on the descriptions.

**Review by c-py — APPROVED** [review]
LGTM! Nice work

**Pull request merged**